src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,173 @@
-            throw new ProprietaireNotFoundException();
-        }
-
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if (uuid == null){
-            uuid = UUID.randomUUID().toString();
-            tokenCacheService.addToCache(proprietaireDTO.getId() ,uuid);
-        }
-
-        String token = jwtHelper.generateToken(email, uuid);
-
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
-
+    if (proprietaireDTO != null && !proprietaireService.verifyPassword(password,
+        proprietaireDTO.getId())) {
+      throw new ProprietaireNotFoundException();
+    } else if (administrateurDTO != null && !administrateurService.verifyPassword(password,
+        administrateurDTO.getId())) {
+      throw new AdministrateurNotFoundException();
     }
 
-    public AuthLoginResponseDTO authProprietaire(@NonNull String token){
-
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-        
-        return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
-        );
-
+    String uuid = tokenCacheService.getFromCache(id);
+    if (uuid == null) {
+      uuid = UUID.randomUUID().toString();
+      tokenCacheService.addToCache(id, uuid);
     }
 
-    public void logoutProprietaire(@NonNull String token){
-
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
+    String token = jwtHelper.generateToken(email, uuid, isAdmin);
+
+    return new AuthLoginResponseDTO(id, token, isAdmin);
+  }
+
+  /**
+   * Authentifie un utilisateur à partir d'un token.
+   *
+   * @param token Le token JWT fourni à l'utilisateur après une connexion réussie.
+   * @return Un objet {@link AuthLoginResponseDTO} contenant l'ID de l'utilisateur, le token, et si l'utilisateur est un administrateur.
+   * @throws TokenExpireOrInvalidException Si le token est expiré ou invalide.
+   */
+  public AuthLoginResponseDTO auth(@NonNull String token) {
+
+    Entry<Long, Boolean> entry = verifyToken(token, false);
+
+    return new AuthLoginResponseDTO(entry.getKey(), token, entry.getValue());
+
+  }
+
+  /**
+   * Déconnecte un utilisateur en invalidant le token actuel et en générant un nouveau.
+   *
+   * @param token Le token JWT de l'utilisateur à déconnecter.
+   */
+  public void logout(@NonNull String token) {
+
+    Entry<Long, Boolean> entry = verifyToken(token, false);
+
+    tokenCacheService.addToCache(entry.getKey(), UUID.randomUUID().toString());
+
+  }
+
+  /**
+   * Vérifie la validité du token JWT, en vérifiant également si l'utilisateur devrait être un administrateur.
+   *
+   * @param token Le token JWT à vérifier.
+   * @param shouldBeAdmin Booléen indiquant si l'utilisateur doit être vérifié en tant qu'administrateur.
+   * @return Une {@link Entry} contenant l'ID de l'utilisateur et un booléen indiquant s'il est administrateur.
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 111).